lib/core/presentation/pages/chat/chat_list/views/check_guild_room_roles_bottomsheet.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:app/core/application/chat/check_guild_room_roles_bloc/check_guild_room_roles_bloc.dart';
+import 'package:app/core/domain/chat/entities/guild_room.dart';
+import 'package:app/core/presentation/pages/chat/chat_list/views/widgets/guild_roles_list.dart';
+import 'package:app/core/presentation/widgets/common/button/linear_gradient_button_widget.dart';
+import 'package:app/core/service/wallet/wallet_connect_service.dart';
+import 'package:app/gen/fonts.gen.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:app/theme/color.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:app/theme/typo.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class CheckGuildRoomRolesBottomSheet extends StatelessWidget {
+  final Function() onEnterChannel;
+  final GuildRoom guildRoom;
+
+  const CheckGuildRoomRolesBottomSheet({
+    super.key,
+    required this.guildRoom,
+    required this.onEnterChannel,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final userWalletAddress =
```

Comment:
how can we assume that we already have wallet address here ? what if the user didnt connect to the wallet ?